### PR TITLE
2.3.x.pgraster

### PR DESCRIPTION
**Pull request by pramsey**
This PR doesn't change configuration or parameters at all, it just addresses [two bugs](https://github.com/mapnik/mapnik/issues/2661) filed against mapnik that affect using mapnik to render pgraster inputs.
- rendering of single-band inputs as grey-scale did not correctly make nodata pixels transparent in the output
- rendering of 16-bit single-band inputs displayed a wrap-around effect

### CODE REVIEW

File: plugins/input/pgraster/pgraster_wkb_reader.cpp [hunk 1]
```diff
@@ -193,8 +193,9 @@ void read_data_band(mapnik::raster_ptr raster,
 
   float* data = (float*)image.getBytes();
   double val;
-  val = reader(); // nodata value, need to read anyway
-  if ( hasnodata ) raster->set_nodata(val);
+  double nodataval;
+  nodataval = reader(); // nodata value, need to read anyway
+  if ( hasnodata ) raster->set_nodata(nodataval);
```
Comment (rafatower) [inline plugins/input/pgraster/pgraster_wkb_reader.cpp]: I prefer how it is done some lines bellow, seems more clear.

``` c++
  val = reader();
  if ( hasnodata ) {
      raster->set_nodata(val); // this doesn't actually *do* anything, just sets a value on the raster object
      nodataval = val;
  }
```

but anyway, it is functionally equivalent to the original code, isn't it?

File: plugins/input/pgraster/pgraster_wkb_reader.cpp [hunk 2]
```diff
@@ -193,8 +193,9 @@ void read_data_band(mapnik::raster_ptr raster,
...
   for (int y=0; y<height; ++y) {
     for (int x=0; x<width; ++x) {
       val = reader();
```
Comment (rafatower) [inline plugins/input/pgraster/pgraster_wkb_reader.cpp:201]: no alpha there, right?

**Comment by rafatower** [comment]
:+1:

File: plugins/input/pgraster/pgraster_wkb_reader.cpp (hunk 2 above)
Comment (pramsey) [inline plugins/input/pgraster/pgraster_wkb_reader.cpp:201]: Right, it's a "data" raster, and any nodata transparency will have to be added by the rastercolorizer in the rendering engine itself.

**Comment by rochoa** [comment]
LGTM.

And no longer `Start with plain white`? :smile:

**Comment by pramsey** [comment]
Actually, thats a really fair question (the "start with plain white" thing) and I'm going to devote a bit of time to answering it and figuring it out now, instead of waiting for a user ticket -- the use of `image.set()` is a bit arbitrary in the code base, to say the least, and I'd like to determine both (a) what it really does and (b) what the best use of it is in the different output types (data, grey, colour).

**Pull request merged**